apps/frontend/src/components/TimeCardPage/EvaluationModal.tsx

```diff
@@ -0,0 +1,182 @@
+    null
+  );
+
+  interface WeeklyEvaluationModalProps {
+    setWeeklyEvaluation: Function;
+    weeklyEvaluation: CommentSchema[];
+  }
+
+  // set value for existing comment
+  useEffect(() => {
+    if (commentToEdit) {
+      setEditedComment(commentToEdit.Content);
+    } else {
+      setEditedComment("");
+    }
+  }, [commentToEdit]);
+
+  const options = [
+    {
+      value: "option1",
+      content: "This associate was a total rock star this week!",
+    },
+    { value: "option2", content: "This associate did just okay this week" },
+    {
+      value: "option3",
+      content: "Lack of professionalism and/or poor behavior",
+    },
+    { value: "option4", content: "N/A" },
+    { value: "option5", content: otherText },
+  ];
+
+  const [user, setUser] = useState<UserSchema>();
+
+  const closeCommentForm = () => {
+    setIsOpenCommentForm(false);
+    onClose();
+  };
+
+  const handleOptionChange = (value: string) => {
+    if (commentToEdit) {
+      const selectedOptionContent = options.find(
+        (opt) => opt.value === value
+      )?.content;
+      setEditedComment(selectedOptionContent);
+      setSelectedOption(value);
+    } else {
+      setSelectedOption(value);
+    }
+  };
+
+  const handleInputOption = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = event.target.value;
+    if (commentToEdit) {
+      setEditedComment(newValue);
+    } else {
+      setOtherText(event.target.value);
+    }
+  };
+
+  const handleSubmit = () => {
+    if (commentToEdit) {
+      const updatedComment = { ...commentToEdit, Content: editedComment };
+      onCommentSubmit(updatedComment);
+      closeCommentForm();
+    } else {
+      let newCommentContent = selectedOption;
+      if (selectedOption === "option5") {
+        newCommentContent = otherText;
+      } else {
+        newCommentContent = options.find(
+          (opt) => opt.value === selectedOption
+        )?.content;
+      }
+
+      const newComment = createNewComment(
+        user,
+        CommentType.Comment,
+        newCommentContent
+      );
+      onCommentSubmit(newComment);
+    }
+  };
+
+  const handleDelete = () => {
+    if (commentToEdit) {
+      const updatedComments = commentList.filter(
+        (comment) => comment !== commentToEdit
+      );
+      setCommentList(updatedComments);
+      onCommentDelete(commentToEdit);
+      setSelectedComment(null);
+      closeCommentForm();
+    }
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={closeCommentForm}>
+      <ModalContent>
+        <ModalHeader> Weekly Performance Evaluation</ModalHeader>
+        <ModalBody>
+          <form onSubmit={handleSubmit}>
+            <FormLabel>
+              How would you describe this associate's overall work performance
+              this week?
+            </FormLabel>
+            <RadioGroup onChange={handleOptionChange} value={selectedOption}>
+              <Stack spacing={2}>
+                <Radio value="option1">
```

Comment:
It seems like these are the same text as what's been defined in the `options` array above. Just so that there's a single source of these text values, would it be possible to have the text here be pulled from the `options` array rather than hard-coded?

i.e. generate the `<Radio />` components in a loop or a map of the `options` array